src/libs/auth.ts

```diff
@@ -4,22 +4,43 @@
 
 import { HTTP } from '../http';
 import { MyMICDSOptions } from '../options';
-import { Observable } from 'rxjs/Observable'; // tslint:disable-line
+
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
 import { tap } from 'rxjs/operators';
 
+import { decode } from 'jsonwebtoken';
+
 export class AuthAPI {
 
-	constructor(private http: HTTP, private options: MyMICDSOptions) { }
+	auth$: BehaviorSubject<AuthSnapshot>;
+	authSnapshot: AuthSnapshot;
+
+	constructor(private http: HTTP, private options: MyMICDSOptions) {
+		this.authSnapshot = { jwt: undefined };
```

Comment:
Maybe here you want to call the JWT getter to see if user is already logged in via `rememberMe`?

Reply:
alright just fixed